### PR TITLE
Setting up associations

### PR DESCRIPTION
**Pull request by kazim110**
This pull request adds the implementation for the `Classroom` and `Student` classes with their respective relationships, addressing the following requirements:

- Created the `Classroom` class with an `@label` instance variable and a has-many relationship with `Student`.
- Implemented the belongs-to relationship between `Student` and `Classroom`.
- Added the `add_student` method to the `Classroom` class, which not only adds a student to the classroom but also sets the classroom for the student.
- Modified the `Student` class to include a `classroom` attribute to represent the classroom to which a student belongs.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
```
Comment (Qoosim) [inline student.rb]: - [ ] Great work setting association between student and classroom. There's little thing missing here, When setting the classroom for a student it also adds it to the classrooms' students. There should be a method that would serve as a `belong-to` side (i.e a student belong to a classroom). You can check this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`) for more details. It should look something similart to this :point_down: 

```
class Student < Person
  attr_reader :classroom
  .
  .

  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end

```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
```
Comment (Qoosim) [inline book.rb]: - [ ] Great job so far. However, there're few things to add here. There should be many-to-many relationship between `Person` and `Book` using the intermediate class `Rental`.
- A book has many rental
Something similar to this :point_down: 

```
  class Book
  
  # *Getters and Setters for @rentals should be present
  attr_accessor :title, :author, :rentals   
 (check this link: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor) for more info.
  .
  .

  def initialize(title, author)
    .
    .
    # *@rentals should be initialized. You can check this link: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars
    @rentals = []
  end
  
  # *A method for adding rentals should be present. You can check this link too: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship
  def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
end

```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (Qoosim) [inline rental.rb]: - [OPTIONAL] It is adviseable to use shortcut for setters and getters. You can check this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor`) for more info.
- [ ]  Getters and Setters for @book and @person instance variables should be present

```
  attr_accessor :date, :book, :person

```

File: person.rb [hunk 4]
```diff
@@ -1,13 +1,14 @@
 class Person < Nameable
-  attr_reader :id
-  attr_accessor :name, age
+  attr_reader :id, :rentals
+  attr_accessor :name, :age
```
Comment (Qoosim) [inline person.rb]: - [OPTIONAL] It is adviseable to use shortcut for setters and getters. You can check this [article](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor`) for more info.
- [ ]  Getters and Setters for @rentals should be present

```
  attr_accessor :name, :age, :rentals

```

File: person.rb [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (Qoosim) [inline person.rb]: - [ ]  A person has many rentals :point_down:   
- A method for adding rentals should be present. You can check [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more info.
```
  def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end

```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @kazim110, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @kazim110 , 

Great job so far!  

May I highlight the following: 
- [x]  Well done implementing all the changes required by the previews TSE👏  ✔️ 
- [x] Your code is clean and well documented✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**